code/game/gamemodes/objectives/basic/steal.dm

```diff
@@ -8,6 +8,20 @@ GLOBAL_LIST_EMPTY(possible_items)
 /datum/objective/steal/get_target()
 	return steal_target
 
+/datum/objective/steal/get_tracking_target(atom/source)
+	var/closest = INFINITY
+	var/turf/source_turf = get_turf(source)
+	var/atom/tracked = null
+	for (var/atom/target in get_trackables_by_type(steal_target, TRUE))
+		var/turf/target_turf = get_turf(target)
+		// Prioritise things that are on the same z
+		var/dist = get_dist(target, source) + abs(source_turf.z - target_turf.z) * 1000
```

Comment:
use `compare_z(here, there)` and if it returns FALSE, give it a massive weight. As long as these are in the same z category, we don't have to put a such critical value.

Reply:
Oh, I can use this to make it not trackable if its on another objective. Thanks for pointing this proc out